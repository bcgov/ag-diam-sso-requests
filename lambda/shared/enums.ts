export const EVENTS = {
  REQUEST_PR_SUCCESS: 'request-pr-success',
  REQUEST_PR_FAILURE: 'request-pr-failure',
  REQUEST_PLAN_SUCCESS: 'request-plan-success',
  REQUEST_PLAN_FAILURE: 'request-plan-failure',
  REQUEST_APPLY_SUCCESS: 'request-apply-success',
  REQUEST_APPLY_FAILURE: 'request-apply-failure',
  REQUEST_CREATE_SUCCESS: 'request-create-success',
  REQUEST_CREATE_FAILURE: 'request-create-failure',
  REQUEST_UPDATE_SUCCESS: 'request-update-success',
  REQUEST_UPDATE_FAILURE: 'request-update-failure',
  REQUEST_DELETE_SUCCESS: 'request-delete-success',
  REQUEST_DELETE_FAILURE: 'request-delete-failure',
  EMAIL_SUBMISSION_FAILURE: 'email-submission-failure',
  REQUEST_LIMIT_REACHED: 'request-limit-reached',
};

export const EMAILS = {
  CREATE_INTEGRATION_SUBMITTED: 'create-integration-submitted',
  CREATE_INTEGRATION_SUBMITTED_BCEID_NONPROD_IDIM: 'create-integration-submitted-bceid-nonprod-idim',
  CREATE_INTEGRATION_SUBMITTED_BCEID_PROD: 'create-integration-submitted-bceid-prod',
  CREATE_INTEGRATION_APPROVED: 'create-integration-approved',
  UPDATE_INTEGRATION_SUBMITTED: 'update-integration-submitted',
  UPDATE_INTEGRATION_APPROVED: 'update-integration-approved',
  BCEID_PROD_APPROVED: 'bceid-prod-approved',
  DELETE_INTEGRATION_SUBMITTED: 'delete-integration-submitted',
  DELETE_INTEGRATION_SUBMITTED_BCEID: 'delete-integration-submitted-bceid',
  TEAM_INVITATION: 'team-invitation',
  TEAM_MEMBER_DELETED_ADMINS: 'team-member-deleted-admins',
  TEAM_MEMBER_DELETED_USER_REMOVED: 'team-member-deleted-user-removed',
  TEAM_DELETED: 'team-deleted',
  REQUEST_LIMIT_EXCEEDED: 'request-limit-exceeded',
  TEAM_API_SERVICE_ACCOUNT_REQUESTED: 'team-api-service-account-requested',
  TEAM_API_SERVICE_ACCOUNT_CREATED: 'team-api-service-account-created',
};
