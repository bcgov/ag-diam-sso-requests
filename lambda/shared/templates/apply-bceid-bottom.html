<strong>* BCeID production integration is not approved and not ready to use yet</strong>
