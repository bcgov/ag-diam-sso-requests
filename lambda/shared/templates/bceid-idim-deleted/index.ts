import fs from 'fs';
import Handlebars from 'handlebars';
import { processRequest } from '../helpers';

const SUBJECT_TEMPLATE = `BCeID Integration Request ID {{request.id}} Deleted`;
const template = fs.readFileSync(__dirname + '/template.html', 'utf8');

const subjectHandler = Handlebars.compile(SUBJECT_TEMPLATE, { noEscape: true });
const bodyHandler = Handlebars.compile(template, { noEscape: true });

export const render = (originalData) => {
  const { request, appUrl } = originalData;
  const data = { request: processRequest(request), appUrl };
  return {
    subject: subjectHandler(data),
    body: bodyHandler(data),
  };
};

export default { render };
