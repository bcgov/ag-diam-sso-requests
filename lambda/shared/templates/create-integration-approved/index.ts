import * as fs from 'fs';
import Handlebars = require('handlebars');
import { processRequest } from '../helpers';
import { Data } from '@lambda-shared/interfaces';
import { sendEmail } from '@lambda-shared/utils/ches';
import { SSO_EMAIL_ADDRESS } from '@lambda-shared/local';
import { getIntegrationEmails } from '../helpers';
import type { RenderResult } from '../index';

const SUBJECT_TEMPLATE = `Pathfinder SSO request approved`;
const template = fs.readFileSync(__dirname + '/template.html', 'utf8');

const subjectHandler = Handlebars.compile(SUBJECT_TEMPLATE, { noEscape: true });
const bodyHandler = Handlebars.compile(template, { noEscape: true });

interface DataProps {
  integration: Data;
}

export const render = async (originalData: DataProps): Promise<RenderResult> => {
  const { integration } = originalData;
  const data = { ...originalData, integration: await processRequest(integration) };

  return {
    subject: subjectHandler(data),
    body: bodyHandler(data),
  };
};

export const send = async (data: DataProps, rendered: RenderResult) => {
  const { integration } = data;
  const emails = await getIntegrationEmails(integration);

  return sendEmail({
    to: emails,
    cc: [SSO_EMAIL_ADDRESS],
    ...rendered,
  });
};

export default { render, send };
