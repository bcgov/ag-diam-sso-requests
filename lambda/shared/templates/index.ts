import * as fs from 'fs';
import Handlebars = require('handlebars');
import { noop } from 'lodash';
import { models } from '@lambda-shared/sequelize/models/models';
import { EVENTS, EMAILS } from '@lambda-shared/enums';
import bceidProdApproved from './bceid-prod-approved';
import createIntegrationApproved from './create-integration-approved';
import createIntegrationSubmitted from './create-integration-submitted';
import createIntegrationSubmittedBceidNonprodIdim from './create-integration-submitted-bceid-nonprod-idim';
import createIntegrationSubmittedBceidProd from './create-integration-submitted-bceid-prod';
import deleteIntegrationSubmitted from './delete-integration-submitted';
import deleteIntegrationSubmittedBceid from './delete-integration-submitted-bceid';
import requestLimitExceeded from './request-limit-exceeded';
import teamDeleted from './team-deleted';
import teamInvitation from './team-invitation';
import teamMemberDeletedAdmins from './team-member-deleted-admins';
import teamMemberDeletedUserRemoved from './team-member-deleted-user-removed';
import updateIntegrationApproved from './update-integration-approved';
import updateIntegrationSubmitted from './update-integration-submitted';
import createTeamApiAccountSubmitted from './create-team-api-account-approved';
import createTeamApiAccountApproved from './create-team-api-account-approved';
import deleteTeamApiAccountSubmitted from './delete-team-api-account-submitted';

const APP_URL = process.env.APP_URL || 'http://localhost:3000';
const API_URL = process.env.API_URL || 'http://localhost:8080/app';
const APP_ENV = process.env.APP_ENV || 'development';
const footer = fs.readFileSync(__dirname + '/footer.html', 'utf8');

Handlebars.registerPartial('footer', footer);

const getBuilder = (key: string) => {
  let builder = { render: noop, send: noop };

  switch (key) {
    case EMAILS.BCEID_PROD_APPROVED:
      builder = bceidProdApproved;
      break;
    case EMAILS.CREATE_INTEGRATION_APPROVED:
      builder = createIntegrationApproved;
      break;
    case EMAILS.CREATE_INTEGRATION_SUBMITTED:
      builder = createIntegrationSubmitted;
      break;
    case EMAILS.CREATE_INTEGRATION_SUBMITTED_BCEID_NONPROD_IDIM:
      builder = createIntegrationSubmittedBceidNonprodIdim;
      break;
    case EMAILS.CREATE_INTEGRATION_SUBMITTED_BCEID_PROD:
      builder = createIntegrationSubmittedBceidProd;
      break;
    case EMAILS.DELETE_INTEGRATION_SUBMITTED:
      builder = deleteIntegrationSubmitted;
      break;
    case EMAILS.DELETE_INTEGRATION_SUBMITTED_BCEID:
      builder = deleteIntegrationSubmittedBceid;
      break;
    case EMAILS.TEAM_DELETED:
      builder = teamDeleted;
      break;
    case EMAILS.REQUEST_LIMIT_EXCEEDED:
      builder = requestLimitExceeded;
      break;
    case EMAILS.TEAM_INVITATION:
      builder = teamInvitation;
      break;
    case EMAILS.TEAM_MEMBER_DELETED_ADMINS:
      builder = teamMemberDeletedAdmins;
      break;
    case EMAILS.TEAM_MEMBER_DELETED_USER_REMOVED:
      builder = teamMemberDeletedUserRemoved;
      break;
    case EMAILS.UPDATE_INTEGRATION_APPROVED:
      builder = updateIntegrationApproved;
      break;
    case EMAILS.UPDATE_INTEGRATION_SUBMITTED:
      builder = updateIntegrationSubmitted;
      break;
    case EMAILS.CREATE_TEAM_API_ACCOUNT_SUBMITTED:
      builder = createTeamApiAccountSubmitted;
      break;
    case EMAILS.CREATE_TEAM_API_ACCOUNT_APPROVED:
      builder = createTeamApiAccountApproved;
      break;
    case EMAILS.DELETE_TEAM_API_ACCOUNT_SUBMITTED:
      builder = deleteTeamApiAccountSubmitted;
      break;
    default:
      break;
  }

  return builder;
};

export interface RenderResult {
  code?: string;
  subject: string;
  body: string;
}

export const renderTemplate = async (code: string, data: any): Promise<RenderResult> => {
  data.appUrl = APP_URL;
  data.apiUrl = API_URL;
  const prefix = APP_ENV === 'production' ? '' : '[DEV] ';
  const builder = getBuilder(code);
  const result = await builder.render(data);

  if (prefix) result.subject = `${prefix}${result.subject}`;
  result.code = code;
  return result;
};

const createEvent = async (data) => {
  try {
    await models.event.create(data);
  } catch (err) {
    console.error(err);
  }
};

export const sendTemplate = async (code: string, data: any) => {
  try {
    const builder = getBuilder(code);
    const rendered = await renderTemplate(code, data);
    await builder.send(data, rendered);
  } catch (err) {
    console.error(err);

    if (data.integration) {
      createEvent({
        eventCode: EVENTS.EMAIL_SUBMISSION_FAILURE,
        requestId: data.integration.id,
        details: { emailCode: code, error: err.message || err },
      });
    }
  }
};

export const sendTemplates = async (emails: { code: string; data: any }[]) => {
  await emails.map((email) => sendTemplate(email.code, email.data));
};

export default { renderTemplate, sendTemplate };
