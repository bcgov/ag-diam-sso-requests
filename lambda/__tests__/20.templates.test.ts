import { renderTemplate } from '@lambda-shared/templates';
import { EMAILS } from '@lambda-shared/enums';
import { formDataDev, formDataProd } from './fixtures';

describe('Email template snapshots', () => {
  it('Should return the expeted email for CREATE_INTEGRATION_SUBMITTED', async () => {
    const rendered = await renderTemplate(EMAILS.CREATE_INTEGRATION_SUBMITTED, { integration: formDataDev });
    expect(rendered.subject).toMatchSnapshot();
    expect(rendered.body).toMatchSnapshot();
  });

  it('Should return the expeted email for CREATE_INTEGRATION_SUBMITTED_BCEID_NONPROD_IDIM', async () => {
    const rendered = await renderTemplate(EMAILS.CREATE_INTEGRATION_SUBMITTED_BCEID_NONPROD_IDIM, {
      integration: formDataDev,
    });
    expect(rendered.subject).toMatchSnapshot();
    expect(rendered.body).toMatchSnapshot();
  });

  it('Should return the expeted email for CREATE_INTEGRATION_SUBMITTED_BCEID_PROD', async () => {
    const rendered = await renderTemplate(EMAILS.CREATE_INTEGRATION_SUBMITTED_BCEID_PROD, {
      integration: formDataProd,
    });
    expect(rendered.subject).toMatchSnapshot();
    expect(rendered.body).toMatchSnapshot();
  });

  it('Should return the expeted email for CREATE_INTEGRATION_APPROVED', async () => {
    const rendered = await renderTemplate(EMAILS.CREATE_INTEGRATION_APPROVED, { integration: formDataDev });
    expect(rendered.subject).toMatchSnapshot();
    expect(rendered.body).toMatchSnapshot();
  });

  it('Should return the expeted email for UPDATE_INTEGRATION_SUBMITTED', async () => {
    const rendered = await renderTemplate(EMAILS.UPDATE_INTEGRATION_SUBMITTED, { integration: formDataDev });
    expect(rendered.subject).toMatchSnapshot();
    expect(rendered.body).toMatchSnapshot();
  });

  it('Should return the expeted email for UPDATE_INTEGRATION_APPROVED', async () => {
    const rendered = await renderTemplate(EMAILS.UPDATE_INTEGRATION_APPROVED, { integration: formDataDev });
    expect(rendered.subject).toMatchSnapshot();
    expect(rendered.body).toMatchSnapshot();
  });

  it('Should return the expeted email for BCEID_PROD_APPROVED', async () => {
    const rendered = await renderTemplate(EMAILS.BCEID_PROD_APPROVED, { integration: formDataDev });
    expect(rendered.subject).toMatchSnapshot();
    expect(rendered.body).toMatchSnapshot();
  });

  it('Should return the expeted email for DELETE_INTEGRATION_SUBMITTED', async () => {
    const rendered = await renderTemplate(EMAILS.DELETE_INTEGRATION_SUBMITTED, { integration: formDataDev });
    expect(rendered.subject).toMatchSnapshot();
    expect(rendered.body).toMatchSnapshot();
  });

  it('Should return the expeted email for DELETE_INTEGRATION_SUBMITTED_BCEID', async () => {
    const rendered = await renderTemplate(EMAILS.DELETE_INTEGRATION_SUBMITTED_BCEID, { integration: formDataDev });
    expect(rendered.subject).toMatchSnapshot();
    expect(rendered.body).toMatchSnapshot();
  });

  it('Should return the expeted email for TEAM_INVITATION', async () => {
    const rendered = await renderTemplate(EMAILS.TEAM_INVITATION, {
      email: 'teat@bcgov.bc.ca',
      team: { id: 1, name: 'testteam' },
      invitationLink: `https://mysite.com/invitation-link`,
    });
    expect(rendered.subject).toMatchSnapshot();
    expect(rendered.body).toMatchSnapshot();
  });

  it('Should return the expeted email for TEAM_MEMBER_DELETED_ADMINS', async () => {
    const rendered = await renderTemplate(EMAILS.TEAM_MEMBER_DELETED_ADMINS, {
      user: { id: 1, displayName: 'test user' },
      team: { id: 1, name: 'testteam' },
    });
    expect(rendered.subject).toMatchSnapshot();
    expect(rendered.body).toMatchSnapshot();
  });

  it('Should return the expeted email for TEAM_MEMBER_DELETED_USER_REMOVED', async () => {
    const rendered = await renderTemplate(EMAILS.TEAM_MEMBER_DELETED_USER_REMOVED, {
      user: { id: 1, displayName: 'test user' },
      team: { id: 1, name: 'testteam' },
    });
    expect(rendered.subject).toMatchSnapshot();
    expect(rendered.body).toMatchSnapshot();
  });

  it('Should return the expeted email for TEAM_DELETED', async () => {
    const rendered = await renderTemplate(EMAILS.TEAM_DELETED, { team: { id: 1, name: 'testteam' } });
    expect(rendered.subject).toMatchSnapshot();
    expect(rendered.body).toMatchSnapshot();
  });

  it('Should return the expeted email for REQUEST_LIMIT_EXCEEDED', async () => {
    const rendered = await renderTemplate(EMAILS.REQUEST_LIMIT_EXCEEDED, { user: { id: 1, displayName: 'test user' } });
    expect(rendered.subject).toMatchSnapshot();
    expect(rendered.body).toMatchSnapshot();
  });
});
