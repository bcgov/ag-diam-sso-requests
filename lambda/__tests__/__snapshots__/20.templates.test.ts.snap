// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email template snapshots Should return the expeted email for BCEID_PROD_APPROVED 1`] = `"[DEV] BCeID Request ID 1 approved and being processed"`;

exports[`Email template snapshots Should return the expeted email for BCEID_PROD_APPROVED 2`] = `
"<h1>Hello Pathfinder Friend,</h1>
<p>
  Your Pathfinder SSO production environment for BCeID <strong>integration request ID 1</strong> has
  been approved.
</p>
<p>
  Your resources will be available in the next 20 minutes or so, please log in to your dashboard to see the progress.
</p>
<p><strong>Project Name: </strong>test</p>
<p><strong>Submitted By: </strong>SSO Admin</p>
<p>If you are not the requester, this email serves only to notify you of the request status.</p>
<p>
  If you have any questions, please contact us by
  <a href=\\"https://chat.developer.gov.bc.ca/channel/sso\\" target=\\"_blank\\" title=\\"Rocket Chat\\" rel=\\"noreferrer\\">
    Rocket.Chat
  </a>
  or email at:
  <a href=\\"mailto:bcgov.sso@gov.bc.ca\\" title=\\"Pathfinder SSO\\" target=\\"_blank\\" rel=\\"noreferrer\\"> bcgov.sso@gov.bc.ca </a>
</p>
<p>Thank you.<br />Pathfinder SSO Team</p>
"
`;

exports[`Email template snapshots Should return the expeted email for CREATE_INTEGRATION_APPROVED 1`] = `"[DEV] Pathfinder SSO request approved"`;

exports[`Email template snapshots Should return the expeted email for CREATE_INTEGRATION_APPROVED 2`] = `
"<h1>Your Pathfinder SSO request ID 1 is approved.</h1>
<p>
  <strong>Project name: </strong>test<br />
  <strong>Submitted by: </strong>SSO Admin
</p>
<p>
  Please <a href=\\"http://localhost:3000/my-dashboard/integrations\\" title=\\"Log in\\" target=\\"_blank\\" rel=\\"noreferrer\\">Log in</a>
  to your dashboard to access JSON Client Installation File. If you are not the requester, this email serves only to
  notify you of the request status.
</p>
<p>
  If you have any questions, please contact us by
  <a href=\\"https://chat.developer.gov.bc.ca/channel/sso\\" target=\\"_blank\\" title=\\"Rocket Chat\\" rel=\\"noreferrer\\">
    Rocket.Chat
  </a>
  or email at:
  <a href=\\"mailto:bcgov.sso@gov.bc.ca\\" title=\\"Pathfinder SSO\\" target=\\"_blank\\" rel=\\"noreferrer\\"> bcgov.sso@gov.bc.ca </a>
</p>
<p>Thank you.<br />Pathfinder SSO Team</p>
"
`;

exports[`Email template snapshots Should return the expeted email for CREATE_INTEGRATION_SUBMITTED 1`] = `"[DEV] Pathfinder SSO request submitted"`;

exports[`Email template snapshots Should return the expeted email for CREATE_INTEGRATION_SUBMITTED 2`] = `
"<h1>Your Pathfinder SSO request ID 1 is successfully submitted.</h1>
<p>
  <strong>Project name: </strong>test<br />
  <strong>Submitted by: </strong>SSO Admin
</p>
<p>
  <strong>The expected processing time is 20 minutes</strong><br />
  Once the request is approved, you will receive another email that your JSON Client Installation is ready. If you are
  not the requester, this email serves only to notify you of the request status.
</p>
<p>
  If you have any questions, please contact us by
  <a href=\\"https://chat.developer.gov.bc.ca/channel/sso\\" target=\\"_blank\\" title=\\"Rocket Chat\\" rel=\\"noreferrer\\">
    Rocket.Chat
  </a>
  or email at:
  <a href=\\"mailto:bcgov.sso@gov.bc.ca\\" title=\\"Pathfinder SSO\\" target=\\"_blank\\" rel=\\"noreferrer\\"> bcgov.sso@gov.bc.ca </a>
</p>
<p>Thank you.<br />Pathfinder SSO Team</p>
"
`;

exports[`Email template snapshots Should return the expeted email for CREATE_INTEGRATION_SUBMITTED_BCEID_NONPROD_IDIM 1`] = `"[DEV] New BCeID Request ID 1"`;

exports[`Email template snapshots Should return the expeted email for CREATE_INTEGRATION_SUBMITTED_BCEID_NONPROD_IDIM 2`] = `
"<h1>Hello IDIM Team,</h1>
<p>
  We are notifying you that a new dev and/or test integration request has been submitted. The request details are below:
</p>
<ul>
  <li><strong>Project name: </strong>test</li>
  <li><strong>Accountable person: </strong>Test User</li>
  <li>
    <strong>URIs:</strong>
    <ul>
      <li> Development: https://b </li> 
    </ul>
  </li>
  <li><strong>Identity Providers Required:</strong>idir</li>
</ul>
<p>
  If you have any questions, please contact us by
  <a href=\\"https://chat.developer.gov.bc.ca/channel/sso\\" target=\\"_blank\\" title=\\"Rocket Chat\\" rel=\\"noreferrer\\">
    Rocket.Chat
  </a>
  or email at:
  <a href=\\"mailto:bcgov.sso@gov.bc.ca\\" title=\\"Pathfinder SSO\\" target=\\"_blank\\" rel=\\"noreferrer\\"> bcgov.sso@gov.bc.ca </a>
</p>
<p>Thank you.<br />Pathfinder SSO Team</p>
"
`;

exports[`Email template snapshots Should return the expeted email for CREATE_INTEGRATION_SUBMITTED_BCEID_PROD 1`] = `"[DEV] New BCeID Request ID 1"`;

exports[`Email template snapshots Should return the expeted email for CREATE_INTEGRATION_SUBMITTED_BCEID_PROD 2`] = `
"<h1>Hello Pathfinder SSO friend,</h1>
<p>Thank you for your integration request. Below is a summary of your integration request details.</p>
<ul>
  <li><strong>Project name: </strong>test</li>
  <li><strong>Accountable person: </strong>Test User</li>
  <li>
    <strong>URIs:</strong>
    <ul>
      <li> Development: https://b </li> <li> Test: https://a </li>
    </ul>
  </li>
  <li><strong>Identity Providers Required: </strong>idir</li>
</ul>
<p><strong>Submitted by: </strong>SSO Admin</p>
<p>
  As you've requested an integration with BCeID, any production integration request requires you to work with the
  Identity and Information Management (IDIM) team.
</p>
<ul>
  <li>
    <strong>URIs:</strong>
    <ul>
      <li> Production: https://a </li>
    </ul>
  </li>
</ul>
<h1>Next Steps</h1>
<ol>
  <li>
    <strong>
      On a best effort basis, the BCeID team will endeavour to reach out to you within 2-3 business days to schedule an
      on-boarding meeting.
    </strong>
  </li>
  <li><strong>Please have answers to the questions below, before your meeting with the IDIM team.</strong></li>
</ol>
<ul>
  <li>What is your estimated volume of initial users?</li>
  <li>Do you anticipate your volume of users will grow over the next three years?</li>
  <li>When do you need access to the production environment by?</li>
  <li>When will your end users need access to the production environment?</li>
</ul>
<p>
  If you have any questions, please contact us by
  <a href=\\"https://chat.developer.gov.bc.ca/channel/sso\\" target=\\"_blank\\" title=\\"Rocket Chat\\" rel=\\"noreferrer\\">
    Rocket.Chat
  </a>
  or email at:
  <a href=\\"mailto:bcgov.sso@gov.bc.ca\\" title=\\"Pathfinder SSO\\" target=\\"_blank\\" rel=\\"noreferrer\\"> bcgov.sso@gov.bc.ca </a>
</p>
<p>Thank you.<br />Pathfinder SSO Team</p>
"
`;

exports[`Email template snapshots Should return the expeted email for DELETE_INTEGRATION_SUBMITTED 1`] = `"[DEV] Pathfinder SSO integration ID 1 deleted"`;

exports[`Email template snapshots Should return the expeted email for DELETE_INTEGRATION_SUBMITTED 2`] = `
"<h1>Your Pathfinder SSO request ID 1 is deleted.</h1>
<p>
  <strong>Project name: </strong>test<br />
  <strong>Deleted by: </strong>SSO Admin
</p>
<p>This email serves only to notify you of the request status.</p>
<p>
  To view your integrations please
  <a href=\\"http://localhost:3000/my-dashboard/integrations\\" title=\\"Log in\\" target=\\"_blank\\" rel=\\"noreferrer\\">Log in</a>.
</p>
<p>
  If you have any questions, please contact us by
  <a href=\\"https://chat.developer.gov.bc.ca/channel/sso\\" target=\\"_blank\\" title=\\"Rocket Chat\\" rel=\\"noreferrer\\">
    Rocket.Chat
  </a>
  or email at:
  <a href=\\"mailto:bcgov.sso@gov.bc.ca\\" title=\\"Pathfinder SSO\\" target=\\"_blank\\" rel=\\"noreferrer\\"> bcgov.sso@gov.bc.ca </a>
</p>
<p>Thank you.<br />Pathfinder SSO Team</p>
"
`;

exports[`Email template snapshots Should return the expeted email for DELETE_INTEGRATION_SUBMITTED_BCEID 1`] = `"[DEV] BCeID Integration ID 1 deleted"`;

exports[`Email template snapshots Should return the expeted email for DELETE_INTEGRATION_SUBMITTED_BCEID 2`] = `
"<h1>Hello IDIM Team,</h1>
<p>We are notifying you that we have received a request to delete an integration. The request details are below:</p>
<ul>
  <li><strong>Requester: </strong>SSO Admin</li>
  <li><strong>Project name: </strong>test #(1)</li>
  <li><strong>Accountable person: </strong>Test User</li>
  <li>
    <strong>URIs:</strong>
    <ul>
      <li> Development: https://b </li>  
    </ul>
  </li>
  <li><strong>Identity Providers Required: </strong>idir</li>
</ul>
<p>
  To view your integrations please
  <a href=\\"http://localhost:3000/my-dashboard/integrations\\" title=\\"Log in\\" target=\\"_blank\\" rel=\\"noreferrer\\">Log in</a>.
</p>
<p>
  If you have any questions, please contact us by
  <a href=\\"https://chat.developer.gov.bc.ca/channel/sso\\" target=\\"_blank\\" title=\\"Rocket Chat\\" rel=\\"noreferrer\\">
    Rocket.Chat
  </a>
  or email at:
  <a href=\\"mailto:bcgov.sso@gov.bc.ca\\" title=\\"Pathfinder SSO\\" target=\\"_blank\\" rel=\\"noreferrer\\"> bcgov.sso@gov.bc.ca </a>
</p>
<p>Thank you.<br />Pathfinder SSO Team</p>
"
`;

exports[`Email template snapshots Should return the expeted email for REQUEST_LIMIT_EXCEEDED 1`] = `"[DEV] Pathfinder SSO request limit reached"`;

exports[`Email template snapshots Should return the expeted email for REQUEST_LIMIT_EXCEEDED 2`] = `
"<h1>Request Limit Exceeded</h1>
<p>Pathfinder SSO user <strong>test user (#1)</strong> has exceeded their daily limit</p>
"
`;

exports[`Email template snapshots Should return the expeted email for TEAM_DELETED 1`] = `"[DEV] Team testteam has been removed by a team admin"`;

exports[`Email template snapshots Should return the expeted email for TEAM_DELETED 2`] = `
"<p>testteam has been removed as well as your membership from the team.</p>
<p>
  To view your integrations please
  <a href=\\"http://localhost:3000/my-dashboard/integrations\\" title=\\"Log in\\" target=\\"_blank\\" rel=\\"noreferrer\\">Log in</a>.
</p>
<p>
  If you have any questions, please contact us by
  <a href=\\"https://chat.developer.gov.bc.ca/channel/sso\\" target=\\"_blank\\" title=\\"Rocket Chat\\" rel=\\"noreferrer\\">
    Rocket.Chat
  </a>
  or email at:
  <a href=\\"mailto:bcgov.sso@gov.bc.ca\\" title=\\"Pathfinder SSO\\" target=\\"_blank\\" rel=\\"noreferrer\\"> bcgov.sso@gov.bc.ca </a>
</p>
<p>Thank you.<br />Pathfinder SSO Team</p>
"
`;

exports[`Email template snapshots Should return the expeted email for TEAM_INVITATION 1`] = `"[DEV] Invitation to join testteam"`;

exports[`Email template snapshots Should return the expeted email for TEAM_INVITATION 2`] = `
"<h1>Hello Pathfinder SSO friend,</h1>
<p>You are invited to join <strong>testteam (#1)</strong></p>
<p>
  Please follow this <a href=\\"http://localhost:8080/app/teams/verify?token=https://mysite.com/invitation-link\\">link</a> to view your invitation in the
  dashboard. This link will expire in <strong>2 business days.</strong>
</p>
<p>
  If you think you're receiving this invitation by accident, or have any questions, please contact the SSO Pathfinder
  team by
  <a href=\\"https://chat.developer.gov.bc.ca/channel/sso\\" target=\\"_blank\\" title=\\"Rocket Chat\\" rel=\\"noreferrer\\">
    Rocket.Chat
  </a>
  or reply to this email
</p>
<p>
  If you have any questions, please contact us by
  <a href=\\"https://chat.developer.gov.bc.ca/channel/sso\\" target=\\"_blank\\" title=\\"Rocket Chat\\" rel=\\"noreferrer\\">
    Rocket.Chat
  </a>
  or email at:
  <a href=\\"mailto:bcgov.sso@gov.bc.ca\\" title=\\"Pathfinder SSO\\" target=\\"_blank\\" rel=\\"noreferrer\\"> bcgov.sso@gov.bc.ca </a>
</p>
<p>Thank you.<br />Pathfinder SSO Team</p>
"
`;

exports[`Email template snapshots Should return the expeted email for TEAM_MEMBER_DELETED_ADMINS 1`] = `"[DEV] Team Admin has modified testteam"`;

exports[`Email template snapshots Should return the expeted email for TEAM_MEMBER_DELETED_ADMINS 2`] = `
"<p>A team member <strong>()</strong> has been removed from <strong>testteam</strong>.</p>
<p>
  To view your integrations please
  <a href=\\"http://localhost:3000/my-dashboard/integrations\\" title=\\"Log in\\" target=\\"_blank\\" rel=\\"noreferrer\\">Log in</a>.
</p>
<p>
  If you have any questions, please contact us by
  <a href=\\"https://chat.developer.gov.bc.ca/channel/sso\\" target=\\"_blank\\" title=\\"Rocket Chat\\" rel=\\"noreferrer\\">
    Rocket.Chat
  </a>
  or email at:
  <a href=\\"mailto:bcgov.sso@gov.bc.ca\\" title=\\"Pathfinder SSO\\" target=\\"_blank\\" rel=\\"noreferrer\\"> bcgov.sso@gov.bc.ca </a>
</p>
<p>Thank you.<br />Pathfinder SSO Team</p>
"
`;

exports[`Email template snapshots Should return the expeted email for TEAM_MEMBER_DELETED_USER_REMOVED 1`] = `"[DEV] Team Admin has modified testteam"`;

exports[`Email template snapshots Should return the expeted email for TEAM_MEMBER_DELETED_USER_REMOVED 2`] = `
"<p>A team admin from <strong>testteam</strong> has removed your membership.</p>
<p>
  To view your integrations please
  <a href=\\"http://localhost:3000/my-dashboard/integrations\\" title=\\"Log in\\" target=\\"_blank\\" rel=\\"noreferrer\\">Log in</a>.
</p>
<p>
  If you have any questions, please contact us by
  <a href=\\"https://chat.developer.gov.bc.ca/channel/sso\\" target=\\"_blank\\" title=\\"Rocket Chat\\" rel=\\"noreferrer\\">
    Rocket.Chat
  </a>
  or email at:
  <a href=\\"mailto:bcgov.sso@gov.bc.ca\\" title=\\"Pathfinder SSO\\" target=\\"_blank\\" rel=\\"noreferrer\\"> bcgov.sso@gov.bc.ca </a>
</p>
<p>Thank you.<br />Pathfinder SSO Team</p>
"
`;

exports[`Email template snapshots Should return the expeted email for UPDATE_INTEGRATION_APPROVED 1`] = `"[DEV] Pathfinder SSO change request approved and can be downloaded"`;

exports[`Email template snapshots Should return the expeted email for UPDATE_INTEGRATION_APPROVED 2`] = `
"<h1>Your Pathfinder SSO change for integration ID 1 is approved.</h1>
<p>
  <strong>Project name: </strong>test<br />
  <strong>Submitted by: </strong>SSO Admin
</p>
<p>
  Please
  <a href=\\"http://localhost:3000/my-dashboard/integrations\\" title=\\"Log in\\" target=\\"_blank\\" rel=\\"noreferrer\\">Log in</a>
  to your dashboard to access JSON Client Installation File. If you are not the requester, this email serves only to
  notify you of the request status.
</p>
<p>
  If you have any questions, please contact us by
  <a href=\\"https://chat.developer.gov.bc.ca/channel/sso\\" target=\\"_blank\\" title=\\"Rocket Chat\\" rel=\\"noreferrer\\">
    Rocket.Chat
  </a>
  or email at:
  <a href=\\"mailto:bcgov.sso@gov.bc.ca\\" title=\\"Pathfinder SSO\\" target=\\"_blank\\" rel=\\"noreferrer\\"> bcgov.sso@gov.bc.ca </a>
</p>
<p>Thank you.<br />Pathfinder SSO Team</p>
"
`;

exports[`Email template snapshots Should return the expeted email for UPDATE_INTEGRATION_SUBMITTED 1`] = `"[DEV] Pathfinder SSO change request submitted"`;

exports[`Email template snapshots Should return the expeted email for UPDATE_INTEGRATION_SUBMITTED 2`] = `
"<h1>Your Pathfinder SSO change request for integration ID 1 is successfully submitted.</h1>
<p>
  <strong>Project name: </strong>test<br />
  <strong>Submitted by: </strong>SSO Admin
</p>
<p>
  <strong>The expected processing time is 20 minutes</strong><br />
  Once the request is approved, you will receive another email once your JSON Client Installation is ready. If you are
  not the requester, this email serves only to notify you of the request status.
</p>
<p>
  If you have any questions, please contact us by
  <a href=\\"https://chat.developer.gov.bc.ca/channel/sso\\" target=\\"_blank\\" title=\\"Rocket Chat\\" rel=\\"noreferrer\\">
    Rocket.Chat
  </a>
  or email at:
  <a href=\\"mailto:bcgov.sso@gov.bc.ca\\" title=\\"Pathfinder SSO\\" target=\\"_blank\\" rel=\\"noreferrer\\"> bcgov.sso@gov.bc.ca </a>
</p>
<p>Thank you.<br />Pathfinder SSO Team</p>
"
`;
